src/WebJobs.Script.WebHost/Management/FunctionsSyncManager.cs

```diff
@@ -79,6 +100,22 @@ public FunctionsSyncManager(IConfiguration configuration, IHostIdProvider hostId
             _azureBlobStorageProvider = azureBlobStorageProvider;
         }
 
+        public IExtensionsOptionProvider ExtensionsOptionProvider
```

Comment:
Why the public properties for these? For testing, tests should just inject any instance they require on construction

Reply:
`FunctionSyncManager` reside in the different DI chain. The original branch, you inject `rootServiceProvider` that has `FunctionSyncManager` instance. However, `HostJsonFileConfigurationSource` and `ScriptEnvironmentVariablesConfigurationSource` is not reside in the `rootServiceProvider` instance. So that if I ran your branch, I see the FunctionSyncManager fails because no instance found about `ExtensionsOptionsProvider`.  My solution is adding Property and inject it to the `FunctionSyncManager`.  However, if it is better way exists, I'd happy to change it. 

## Doesn't work
```csharp
services.AddSingleton<ExtensionOptionsProvider>(new ExtensionOptionsProvider(rootServiceProvider, services));
```
## Works
```csharp
                   // Hidrate ExtensionOptionProvider to FunctionSyncManager
                    var functionSyncManager = rootServiceProvider.GetService<IFunctionsSyncManager>();
                    var serviceProvider = services.BuildServiceProvider();
                    functionSyncManager.ExtensionsOptionProvider = new ExtensionsOptionProvider(serviceProvider, services);
```

https://github.com/Azure/azure-functions-host/blob/d7ac580aad39489592f3c64d88d52fb4517d5593/src/WebJobs.Script.WebHost/WebScriptHostBuilderExtension.cs#L157